includes/polling/ports/port-poe.inc.php

```diff
@@ -2,60 +2,76 @@
-    'cpeExtPsePortEntPhyIndex',
-    'cpeExtPsePortPolicingCapable',
-    'cpeExtPsePortPolicingEnable',
-    'cpeExtPsePortPolicingAction',
-    'cpeExtPsePortPwrManAlloc',
-);
-
-$peth_oids = array(
-    'pethPsePortAdminEnable',
-    'pethPsePortPowerPairsControlAbility',
-    'pethPsePortPowerPairs',
-    'pethPsePortDetectionStatus',
-    'pethPsePortPowerPriority',
-    'pethPsePortMPSAbsentCounter',
-    'pethPsePortType',
-    'pethPsePortPowerClassifications',
-    'pethPsePortInvalidSignatureCounter',
-    'pethPsePortPowerDeniedCounter',
-    'pethPsePortOverLoadCounter',
-    'pethPsePortShortCounter',
-    'pethMainPseConsumptionPower',
-);
-
-if ($this_port['dot3StatsIndex'] && $port['ifType'] == 'ethernetCsmacd') {
-    $rrd_name = getPortRrdName($port_id, 'poe');
-    $rrd_def = RrdDefinition::make()
-        ->addDataset('PortPwrAllocated', 'GAUGE', 0)
-        ->addDataset('PortPwrAvailable', 'GAUGE', 0)
-        ->addDataset('PortConsumption', 'DERIVE', 0)
-        ->addDataset('PortMaxPwrDrawn', 'GAUGE', 0);
-
-    $upd = "$polled:".$port['cpeExtPsePortPwrAllocated'].':'.$port['cpeExtPsePortPwrAvailable'].':'.$port['cpeExtPsePortPwrConsumption'].':'.$port['cpeExtPsePortMaxPwrDrawn'];
-
-    $fields = array(
-        'PortPwrAllocated'   => $port['cpeExtPsePortPwrAllocated'],
-        'PortPwrAvailable'   => $port['cpeExtPsePortPwrAvailable'],
-        'PortConsumption'    => $port['cpeExtPsePortPwrConsumption'],
-        'PortMaxPwrDrawn'    => $port['cpeExtPsePortMaxPwrDrawn'],
-    );
-
-    $tags = compact('ifName', 'rrd_name', 'rrd_def');
-    data_update($device, 'poe', $tags, $fields);
-
-    echo 'PoE ';
-}//end if
+if (($device['os'] == 'vrp')) {
+    //Tested against Huawei 5720 access switches
+    if (isset($this_port['hwPoePortEnable'])) {
+        $rrd_name = getPortRrdName($port_id, 'poe');
+        $rrd_def = RrdDefinition::make()
+            ->addDataset('PortPwrAllocated', 'GAUGE', 0)
+            ->addDataset('PortPwrAvailable', 'GAUGE', 0)
+            ->addDataset('PortConsumption', 'GAUGE', 0)
+            ->addDataset('PortMaxPwrDrawn', 'GAUGE', 0);
+        $upd = "$polled:".$this_port['hwPoePortReferencePower'].':'.$this_port['hwPoePortMaximumPower'].':'.$this_port['hwPoePortConsumingPower'].':'.$this_port['hwPoePortPeakPower'];
+
+        $fields = array(
+                'PortPwrAllocated'   => $this_port['hwPoePortReferencePower'],
+                'PortPwrAvailable'   => $this_port['hwPoePortMaximumPower'],
+                'PortConsumption'    => $this_port['hwPoePortConsumingPower'],
+                'PortMaxPwrDrawn'    => $this_port['hwPoePortPeakPower'],
+                   );
+
+        $tags = compact('ifName', 'rrd_name', 'rrd_def');
+        data_update($device, 'poe', $tags, $fields);
+        echo 'PoE(vrp) ';
+    }
+} elseif (($device['os'] == 'ios')||($device['os'] == 'ciscosb')) {
+    // Code for Cisco IOS, tested on 2960X
+    if (isset($this_port['cpeExtPsePortPwrAllocated'])) {
```

Comment:
Where is `$this_port['cpeExtPsePortPwrAllocated']` coming from as it's not in the code anywhere before this?

Reply:
$this_port is defined in line 491 of includes/polling/ports.inc.php as &port_stats[$ifIndex].